arch/arm64/core/reset.c

```diff
@@ -125,6 +125,11 @@ void z_arm64_el2_init(void)
 	write_sctlr_el2(reg);
 
 	reg = read_hcr_el2();
+	/* when EL2 is enable in current security status:
+	 * Clear TGE bit: All exceptions that would not be routed to EL2;
+	 * Clear IMO bit: Physical IRQ interrupts are not taken to EL2 and EL3.
```

Comment:
```suggestion
	 * Clear TGE bit: All exceptions that would be routed to EL2 are routed to EL1;
	 * Clear IMO bit: Physical IRQ interrupts are not taken to EL2 and EL3.
```

I'm a bit lost on the IMO bit. According to the documentation: `If the value of HCR.TGE is 0, then virtual SError interrupts are enabled in Non-secure state.`. Where did you get that comment regarding physical IRQs?

Reply:
In the Arm® Architecture Reference Manual(page D13-3336), the HCR_EL2 register's IMO bit has below feature: when IMO bit is 1, `Physical IRQ interrupts are taken to EL2, unless they are routed to EL3.`. 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/40254934/4e9e3de1-6aed-4374-b050-d4581a413a79)

What you describe may be the HCR_ELR.AMO bit:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/40254934/4f0cc3d0-85bb-45c4-b6b7-a2f5dc7ff1ef)
So, I clear HCR_EL2.IMO bit.

Reply:
Then do we need also to clear the AMO bit?
Also, it would be better to separate the changes in `reset.c` and `cpu.h` into a new commit.

Reply:
Thanks,  the reason why there is no fatal now may be that no SError occurred.  But when zephyr boot using uboot on roc_rk3568_pc, HCR_EL2.AMO bit are set to 1 too.  Clear it may be a good choice. 
On some platforms such as qemu_ cortex_ a53, this operation is not necessary, because the value of HCR_EL2 register is zero. I will add a new commit to annouce it.